packages/ckeditor5-engine/docs/framework/guides/deep-dive/schema.md

```diff
@@ -44,6 +44,48 @@ While this would be incorrect:
 </$root>
 ```
 
+## Declaring as a limit element
+
+Consider a feature like an image caption. The caption text area should construct a boundary to some internal actions.
+
+ - A selection that starts inside should not end outside.
+ - Pressing <kbd>Backspace</kbd> or <kbd>Delete</kbd> should not delete the area. Pressing <kbd>Enter</kbd> should not split the area.
+
+ It should also act as a boundary for external actions:
+
+ - A selection that starts outside, should not end inside.
+
+```js
+schema.register( 'myCaption', {
+    isLimit: true
+} );
+```
+
+{@link module:engine/model/schema~SchemaItemDefinition#isLimit `isLimit`} makes the engine construct such boundaries, and let {@link module:engine/model/utils/selection-post-fixer `selection-post-fixer`} update the user's selection if needed.
+
+<info-box>
+    "Limit element" does not mean "editable element". The concept of "editability" is reserved for the view and expressed by {@link module:engine/view/editableelement~EditableElement `EditableElement` class}.
+</info-box>
+
+## Declaring as a self-sufficient object
+
+For the image caption as in the example above it does not make much sense to select the caption box, then copy or drag it somewhere else.
+A caption without the image it describes does not make much sense. However, the image is more self-sufficient. Most likely users should be able to select the entire image (with all its internals), then copy or move it around. {@link module:engine/model/schema~SchemaItemDefinition#isObject `isObject`} should be used to mark such behavior.
+
+```js
+schema.register( 'myImage', {
+    isObject: true
+} );
+```
+
+Every "object" is also a "limit" element. 
+
+<info-box>
+    It means for every element with `isObject` set to `true`, {@link module:engine/model/schema~Schema#isLimit `schema.isLimit( element )`} will always return `true`. 
+
+    However, {@link module:engine/model/schema~Schema#getDefinition `schema.getDefinition( 'element' )`} may return `false` in a case when `{ isLimit: true, isObject: true}` was registered.
```

Comment:
Given that `Schema#getDefinition()` is only used internally in our codebase and it's unlikely anyone will use it in their integration (maybe it should be private BTW?) I think this clarification is unnecessary, even confusing.

Reply:
I don't agree. This method is part of public API. So it's ok to mention it in the deep dive as this is something that may be confusing.